sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpFluxRequestBody.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.okhttp.implementation;
+
+import com.azure.core.implementation.util.BinaryDataContent;
+import com.azure.core.util.logging.ClientLogger;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okio.BufferedSink;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * An {@link okhttp3.RequestBody} subtype that sends {@link BinaryDataContent}
+ * as {@link Flux} of {@link ByteBuffer} in an unbuffered manner.
+ * This class accepts any {@link BinaryDataContent} as catch-all for backwards compatibility
+ * but ideally should be used only with reactive payloads.
+ */
+public class OkHttpFluxRequestBody extends OkHttpStreamableRequestBody<BinaryDataContent> {
+
+    private static final ClientLogger LOGGER = new ClientLogger(OkHttpFluxRequestBody.class);
+
+    private final AtomicBoolean bodySent = new AtomicBoolean(false);
+    private final OkHttpClient okHttpClient;
+
+    public OkHttpFluxRequestBody(
+        BinaryDataContent content, long effectiveContentLength, MediaType mediaType, OkHttpClient okHttpClient) {
+        super(content, effectiveContentLength, mediaType);
+        this.okHttpClient = Objects.requireNonNull(okHttpClient, "'okHttpClient' cannot be null.");
+    }
+
+    @Override
+    public void writeTo(BufferedSink bufferedSink) throws IOException {
+        if (bodySent.compareAndSet(false, true)) {
+            Mono<BufferedSink> requestSendMono = content.toFluxByteBuffer()
+                .publishOn(Schedulers.boundedElastic())
```

Comment:
I guess this thread jump to Bounded-Elastic-Thread is justified enough? Without this, Flux that is not backed by the user thread will use caller OkHttp-Thread to produce chunks, hence, the potential for using the same OkHttp-Thread to notify IOException downstream. So by using Bounded-Elastic-Thread, the OkHttp-Thread is freed asap (which could be from the bounded IO thread pool) for handling another request.

Not sure there is another reason, does it worth adding a code comment for future reference?